Create 05-review file and addition of possible topics (#23)
empa-scientific-it/how-to-git: src/content/docs/05-review.mdx

---
title: How can changes be reviewed?
slug: review
sidebar:
  order: 5
---

import { Aside } from '@astrojs/starlight/components';

### What are Reviews good for? 
When working together on a project, changes are usually made in branches and checked before merging into the main branch.
Reviews let teammates give feedback, helping you see your work from a fresh perspective and improve it.

> Reviews aren’t just for code; they can also cover documentation, designs, or settings.

Reviewer: ```suggestion
Reviews are not just for code; they can also cover documentation, metadata files, or settings.
```

> They can even be made required to keep quality and consistency high.

Reviewer 2: What does this sentence mean?

Author: @despadam  what sentence do you mean, there are 4

> The common types are code reviews, resp. pull request (PR) reviews, but reviews can also be done on other source of text.

Reviewer: ```suggestion
The common types are code reviews, also known as pull request (PR) reviews, but reviews can also be done on other source of text.
```

In Github projects, reviews can be made obligatory, which can help for supervising changes.

<Aside type="tip">
Be friendly and specific while reviewing: Tell what needs to be changed and how, and give a reason if it is not clear.
</Aside>

### Request a review as an author
A review should be requested, if the work on a topic has finished.

> If the work on a topic has finished, means the branch can be merged to the main branch.

Reviewer: ```suggestion
Finishing the work on a topic means the branch can be merged to the main branch.
```

Reviewer 2: Also add links here to how pull requests are made

> For that a Pull Request is opened and, one can choose a review from someone specific.

Reviewer: ```suggestion
To do this, you open a Pull Request and choose someone specific to review it.
```

> It makes more sense, if the person to review it is not involved in that specific topic, but is close to the it.

Reviewer: ```suggestion
It is often best to choose a reviewer who was not directly involved in that topic but is familiar with the project.
```

> - To create a PR review, you first open a [Pull Request](../docs/03-parallel-topics.mdx#merge-branches).

Reviewer 2: Ah ok the link is actually here, but I think it would be best to move it to the top of the section. Instead of explaining first and then giving the link to that section

> - On the top right you can choose who to assign for the review.

Reviewer: Is this top right or more in the middle of the page? Below reviewers

Author: @fabioacl don't know

> - Now you need to wait for the reviewer to be finished

Reviewer 2: ```suggestion
- Now you just need to wait for the review.
```

### Make changes as a reviewer

As a reviewer you will get a notification (depending on your settings) that you are requested.

> 1. You can start the review by clicking on the green button.

Reviewer 2: Which button?

> 1. Go to a line of text that you would change, click on the plus symbol :heavy_plus_sign: (Remember, we recommend to always only use one Line per sentence [here](../docs/02-track-changes.mdx#change-files))

Reviewer: ```suggestion
2. Go to a line of text that you would change, click on the plus symbol :heavy_plus_sign: (Remember, we recommend to always only use one Line per sentence [here](../docs/02-track-changes.mdx#change-files))
```

> 1. Go to a line of text that you would change, click on the plus symbol :heavy_plus_sign: (Remember, we recommend to always only use one Line per sentence [here](../docs/02-track-changes.mdx#change-files))

Reviewer 2: I think it is a good idea to add some screenshots

3. Either add a comment, or choose to make a suggestion (click on the plusminus symbol), which gives you the option to do changes to a sentence.
  - A suggestion can simply be accepted by the author, where as a comment just gives a hint for a change
4. Click the green button to start the review or add review to previous.

> 5. When you finsished your comments click the green button for "finish your review" on the top

Reviewer: ```suggestion
5. When you are finished with your comments click the green button for "finish your review" on the top
```

>   - This will give a notification to the author for further actions

Reviewer: ```suggestion
  - This will send a notification to the author for further actions
```

### Author handling of Request changes from reviewer
[Description TODO]
- accepting them
- not accepting them

> - adressing them in new issues for later changes?

Reviewer 2: This section is not finished

<Aside type="tip">
[Maybe this should be a Note and not a tip, if it is possible]
Changes and reviews will go back and forth until every party is satisfied with the content.
</Aside>

### Request a change to some part of the text from someone
[Describe TODO]
- Best way is to create an issue, assing it to that creator to ask for changes in certain topic
- person can later create a branche from that issue, create changes and ask you to be the reviewer

> - are there other ways?

Reviewer 2: This section is not finished
